src/graph/unit_graph.cc

```diff
@@ -312,6 +366,10 @@ class UnitGraph::COO : public BaseHeteroGraph {
     return adj_;
   }
 
+  bool IsHypersparse() const {
+    return NumVertices(SrcType()) / 8 > NumEdges(EdgeType());
+  }
```

Comment:
`(NumVertices(SrcType()) + NumVertices(DstType())) / 16 > NumEdges(EdgeType())`? BTW, you could directly use adj_ too but since these calls should be inlined, it should be fine either way.

Reply:
Oh, I see your point here. Look like if |Src| >> |E|, we should restrict COO and CSC. If |Dst| >> |E|, we should restrict COO and CSR. Is that right?

Reply:
Yes.

Reply:
Look like we need a more flexible way to specify the restricted formats. Any idea?